pandas/core/groupby/ops.py

```diff
@@ -567,7 +568,9 @@ def _cython_operation(
         if swapped:
             result = result.swapaxes(0, axis)
 
-        if is_datetime64tz_dtype(orig_values.dtype):
+        if is_datetime64tz_dtype(orig_values.dtype) or is_period_dtype(
```

Comment:
hmm you maybe able to collapse this entire thing to

```
if is_datetimelike:
    result = type(orig_values)(result.astype(np.int64), dtype=orig_values.dtype)
```

(and remove the next case)

Reply:
It looks like this ends up raising in that case: https://github.com/pandas-dev/pandas/blob/master/pandas/tests/groupby/test_groupby.py#L150

Reply:
can you add an issue reference and a short comment on why we are doing this